drakrun/drakrun/draksetup.py

```diff
@@ -83,6 +84,83 @@ def check_root():
         return True
 
 
+def sanity_check():
+    logging.info("Checking xen-detect...")
+    proc = subprocess.run('xen-detect -N', shell=True)
+
+    if proc.returncode != 1:
+        logging.error('It looks like the system is not running on Xen. Please reboot your machine into Xen hypervisor.')
+        return False
+
+    logging.info("Testing if xl tool is sane...")
+
+    try:
+        subprocess.run('xl info', shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=True)
+    except subprocess.CalledProcessError:
+        logging.exception("Failed to test xl command.")
+        return False
+
+    if not perform_xtf():
+        logging.error("Your Xen installation doesn\'t pass the necessary tests.")
+        return False
+
+    return True
+
+
+def perform_xtf():
+    logging.info('Testing your Xen installation...')
+    module_dir = os.path.dirname(os.path.realpath(__file__))
+    cfg_path = os.path.join(module_dir, "tools", "test-hvm64-example.cfg")
+    firmware_path = os.path.join(module_dir, "tools", "test-hvm64-example")
```

Comment:
Is this file supposed to be there in GitHub? I couldn't find it
![image](https://user-images.githubusercontent.com/40723245/109860886-ae7eb980-7c84-11eb-9b01-65a6a3578e81.png)

Reply:
It gets generated while building the package

Reply:
You can generate this binary on your own by building this repo:
https://xenbits.xen.org/gitweb/?p=xtf.git;a=tree;hb=HEAD

it's not included directly here, as @chivay wrote, it's generated during the package build

Reply:
Oh okay, thanks for the link